cmd/lakectl/cmd/plugin_list.go

```diff
@@ -0,0 +1,143 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const pluginListTemplate = `The following lakectl-compatible plugins are available:
+
+{{- range .Plugins }}
+  {{- if or .Verbose .Warning }}
+  {{ .Path }}
+  {{- else }}
+  {{ .Name }}
+  {{- end }}
+  {{- if .Warning }}
+    - {{ "warning:" | yellow }} {{ .Warning }}
+  {{- end }}
+{{- end }}
+`
+
+type pluginInfo struct {
+	Name    string
+	Path    string
+	Warning string
+	Verbose bool
+}
+
+var pluginListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List available lakectl plugins",
+	Long:  `Scans the PATH for executables named "lakectl-*" and lists the detected plugins.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		verbose, _ := cmd.Flags().GetBool("verbose")
+		envPathDirs := filepath.SplitList(os.Getenv("PATH"))
+
+		// Filter out duplicates from envPathDirs
+		var pathDirs []string
+		seen := make(map[string]struct{})
+		for _, dir := range envPathDirs {
+			if dir == "" {
+				// Skip empty entries in PATH
+				continue
+			}
+			if _, ok := seen[dir]; !ok {
+				seen[dir] = struct{}{}
+				pathDirs = append(pathDirs, dir)
+			}
+		}
+
+		// Iterate through pathDirs to build the final list with correct overshadowing
+		pluginsOutput := make([]pluginInfo, 0)
+		foundEffectivePlugins := make(map[string]string) // plugin name -> path of chosen executable
+
+		for _, dir := range pathDirs {
+			files, err := os.ReadDir(dir)
+			if err != nil {
+				// Silently ignore errors from non-existent or unreadable directories in PATH
+				continue
+			}
+
+			for _, file := range files {
+				fileName := file.Name()
+				if !strings.HasPrefix(fileName, "lakectl-") {
+					continue
+				}
+				pluginName := strings.TrimPrefix(fileName, "lakectl-")
+				if pluginName == "" {
+					continue
+				}
+
+				fullPath := filepath.Join(dir, fileName)
+				absPath, err := filepath.Abs(fullPath)
+				if err != nil {
+					// Silently ignore errors from file paths that cannot be resolved
+					continue
+				}
+
+				fileInfo, statErr := file.Info()
+				if statErr != nil {
+					pluginsOutput = append(pluginsOutput, pluginInfo{Name: pluginName, Path: fullPath, Warning: fmt.Sprintf("error getting file info: %v", statErr), Verbose: verbose})
+					continue
+				}
+				if fileInfo.IsDir() {
+					continue
+				}
+
+				var isExec bool
+				if runtime.GOOS == "windows" {
+					// On Windows, check if the file is executable by checking its extension
+					ext := strings.ToLower(filepath.Ext(fileName))
+					isExec = ext == ".exe" || ext == ".com" || ext == ".bat" || ext == ".cmd"
+				} else {
+					// On Unix-like systems, check if the file is executable by user, group, or others
```

Comment:
This is often won't. Just use [access](https://pkg.go.dev/syscall#Access), which is always right _and_ matches what the shell thinks. 
Unfortunately I don't think this works on WIN32; there _access  only knows about read/write status.

Reply:
The problem is access is syscall which if I can avoid in this case, if you check the unix implementation the fallback after calling access that maybe implemented it fallback to check the execution bit. This code will work for all platforms.

Reply:
That "fallback" will never be used.  And it _literally_ calls access on every user-friendly system.

In particular if you do this then _you cannot_ use the PATH lookup of exec*p as I requested: it will be a security bug for both `sudo lakectl my-plugin ...` and/or `lakectl my-plugin-which-is-setuid`, because I can arrange for the listing to show one plugin path but the command actually to run another plugin.